apollo-ios-pagination/Sources/ApolloPagination/AsyncGraphQLQueryPager.swift

```diff
@@ -191,9 +197,10 @@ public class AsyncGraphQLQueryPager<Model>: Publisher {
   /// Subscribe to the results of the pager, with the management of the subscriber being stored internally to the `AnyGraphQLQueryPager`.
   /// - Parameter completion: The closure to trigger when new values come in.
   public func subscribe(completion: @MainActor @escaping (Output) -> Void) {
-    publisher.sink { result in
+    let cancellable = publisher.sink { result in
       Task { await completion(result) }
-    }.store(in: &cancellables)
+    }
+    _ = $cancellables.mutate { $0.insert(cancellable) }
```

Comment:
Does this introduce a possible race condition where the `cancellable` set is cancelled between the time this one is created and the time when it is inserted into the set?

Reply:
It seems unlikely, but I'm also not sure if it's a consequential effect?

This is valid:

```swift
let cancellables: Set<AnyCancellable> = []
let cancellable: AnyCancellable = ... 
cancellable.cancel()
cancellables.insert(cancellable)
```

Reply:
On further examination – I think that the only thing this change really does is remove the guarantee that we're setup to publish values prior to a `subscribe` being called. 

i.e., with code like this:

```swift
let pager = AsyncGraphQLQueryPager(...)
pager.subscribe { ... } 
```

There's a chance the act of a subscriber creating an `AnyCancellable` to be internally managed by the `AsyncGraphQLQueryPager` is executed _prior_ to the `init` setting up to forward results from the `AsyncGraphQLQueryPagerCoordinator`. There's no negative impact to this.

Even if a networking operation somehow begins and completes execution prior to either `subscribe` or `init`'s `Task` completing – its data will still be forwarded the moment that the Task does complete.

Reply:
This may be a better option than what's in `main` currently – especially for edge cases in SwiftUI. I haven't reproduced this, but this is a bit of a thought exercise.

Imagine we have a simple `ViewModel` that instantiates an `AsyncGraphQLQueryPager`, and has functions for `fetch`, `refetch` and `loadNext`:

```swift
@Observable class ViewModel {
  let pager: AsyncGraphQLQueryPager<[String]>!
  var names: [String] = []
  init() {
    Task {
      pager = AsyncGraphQLQueryPager(...)
    }
  }

  func fetch() async {
    await pager.fetch()
  }
  func refetch() async {
    await pager.fetch()
  }
  func loadNext() async {
    try? await pager.loadNext()
  }
}
```

Note the instantiation – we do so in a `Task` within the `init`. 

A common pattern would be to use a top-level task on the top-level view to perform the initial `fetch`:

```swift
struct SwiftUIView: View {
  @StateObject var viewModel = ViewModel()
  var body: some View {
    List($viewModel.names) { name in
      Text(name)
        .task {
          guard name == viewModel.names.last else { return }
          await viewModel.loadNext()
        }
    }
    .task {
      await viewModel.fetch()
    }
    .refreshable {
      await viewModel.refetch()
    }
  }
}
```

There's actually more concern in this existing code than I would have for the proposed code. Specifically: the `task` modifier on the top-level view can be treated as analogous to `viewWillAppear`. What happens if we call `viewModel.fetch()`, and the view model's `init`'s `Task` hasn't completed yet? There would be no pager to fetch from.

The proposed code would move the initialization of a `pager` outside of a `Task`. That means that in a similar scenario to the one above, the pager would begin fetching – and then forward data when the pager's `init` `Task` completes.